packages/extension-chakra-storefront/src/components/with-layout/with-layout.tsx

```diff
@@ -313,28 +328,40 @@ const withLayout = <P extends object>(WrappedComponent: React.ComponentType<P>)
                     </Box>
                     {!isOnline && <OfflineBanner />}
                     <AddToCartModalProvider>
-                        <SkipNavContent
-                            style={{
-                                display: 'flex',
-                                flexDirection: 'column',
-                                flex: 1,
-                                outline: 0
-                            }}
-                        >
+                        {!isNavigationBlocked ? (
+                            <SkipNavContent
+                                style={{
+                                    display: 'flex',
+                                    flexDirection: 'column',
+                                    flex: 1,
+                                    outline: 0
+                                }}
+                            >
+                                <Box
+                                    as="main"
+                                    id="app-main"
+                                    role="main"
+                                    display="flex"
+                                    flexDirection="column"
+                                    flex="1"
+                                >
+                                    <OfflineBoundary isOnline={false}>
+                                        <WrappedComponent {...(props as P)} />
```

Comment:
As mentioned in our direct chat, we might want to use an overlay for the spinner as opposed to controlling what components are rendered. If we do this it means the application and its router and providers all get unmounted and remounted and I believe this is having some unwanted side effects, I think one of those is that the route context is being reset.